src/translate.py

```diff
@@ -502,7 +504,7 @@ class Expression(abc.ABC):
     def dependencies(self) -> List["Expression"]:
         ...
 
-    def use(self) -> None:
+    def use(self, stack_info: StackInfo) -> None:
```

Comment:
The biggest style question I had is if I should *just* pass the `typemap` to `use(...)`, or pass the whole `stack_info`. 
Not sure how gross/abstraction-breaking this is.

Reply:
passing stack_info to use() makes sense to me. Many expressions already include a reference to it, so in principle I think it would also be fine to put it within the Expression base class, but I guess that gets too annoying plumbing-wise